com.hilotec.elexis.messwerte.v2/src/com/hilotec/elexis/messwerte/v2/views/MessungenUebersichtV21.java

```diff
@@ -106,16 +107,27 @@ public String getText(Object element){
 		}
 	};
 	
-	private void setCurPatient(Patient patient){
-		if (patient == null) {
-			form.setText(Messages.MessungenUebersicht_kein_Patient);
-		} else {
-			form.setText(patient.getLabel());
-		}
-		CTabItem tab = tabfolder.getSelection();
-		Control c = tab.getControl();
-		MessungTyp t = (MessungTyp) c.getData(DATA_TYP);
-		refreshContent(patient, t);
+	private void setCurPatient(final Patient patient){
+		Runnable runnable = new Runnable() {
+			public void run() {
+            //multithreading to avoid directly updating the UI which causes Illegal thread access
+				UiDesk.asyncExec(new Runnable() {
+					@Override
+					public void run(){
+						if (patient == null) {
+							form.setText(Messages.MessungenUebersicht_kein_Patient);
+						} else {
+							form.setText(patient.getLabel());
+						}
+						CTabItem tab = tabfolder.getSelection();
+						Control c = tab.getControl();
+						MessungTyp t = (MessungTyp) c.getData(DATA_TYP);
+						refreshContent(patient, t);
+					}
+				});     
+			}
+		};
+		new Thread(runnable).start();
```

Comment:
Ich denke im Zusammenhang mit diesem Problem, solltest du dir die Klasse @ch.elexis.core.ui.eventsElexisUiEventListenerImpl@ ansehen. Die korrekte Lösung liegt denke ich nicht in weiteren separierten Aufrufen von UiDesk.asyncExec sondern der korrekten Anwendung dieses Listeners! 

Das sollte es dir sogar ermöglichen vorhandene UiDesk.asyncExecs zu löschen!